### PR TITLE
[MJAVADOC-825] Prefer NullPointerExceptions for null arguments

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java [hunk 1]
```diff
@@ -534,11 +534,16 @@ protected static JavaVersion getJavadocVersion(File javadocExe)
      * @return the version of the javadoc for the output, only digits and dots
      * @throws PatternSyntaxException if the output doesn't match with the output pattern
      *             {@code (?s).*?[^a-zA-Z]([0-9]+\\.?[0-9]*)(\\.([0-9]+))?.*}.
-     * @throws IllegalArgumentException if the output is null
+     * @throws NullPointerException if the output is null
+     * @throws IllegalArgumentException if the output is empty
      */
     protected static String extractJavadocVersion(String output) throws IllegalArgumentException {
```
Comment (pzygielo) [inline src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java:540]: Given `IAEx` being declared as thrown by, perhaps `NPEx` could also be added for style consistency?
Or `IAEx` declaration removed?

Comment (elharo) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java:540]: Both have @throws clauses. Or did you mean something else?

Comment (pzygielo) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java:540]: `IllegalArgumentException` is `RTEx` (as `NPEx` is). And is declared as being thrown by this method. Thus - let's add `NullPointerException` to the `throws` list.
Or remove `throws`.
Here, and in the second modified method.

Comment (pzygielo) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java:540]: I see this:
```java
protected static String extractJavadocVersion(String output) throws IllegalArgumentException {
```
I'd like to see this instead:
```java
protected static String extractJavadocVersion(String output) throws IllegalArgumentException, NullPointerException {
```
or this:
```java
protected static String extractJavadocVersion(String output) {
```

Comment (elharo) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java]: Good spot. That should go the other way per guidelines. Removed both.

**Review by michael-o — APPROVED** [review]

**Pull request merged**